plugin.sh.tmux

```diff
@@ -14,6 +14,34 @@ config_file="$root_dir/config.yaml"
 plugin_dir="$root_dir/plugin"
 init_file="$plugin_dir/init.tmux"
 
+# XDG
+case "$(tmux show-option -gvq @tmux-which-key-xdg-enable)" in
+    1 | true)
+        if [ -z "$XDG_CONFIG_HOME" ]; then
+            echo "[tmux-which-key] XDG_CONFIG_HOME is not set"
+            exit 1
+        fi
+        if [ ! -d "$XDG_CONFIG_HOME" ]; then
+            echo "[tmux-which-key] XDG_CONFIG_HOME: $XDG_CONFIG_HOME does not exist"
+            exit 1
+        fi
+        if [ -z "$XDG_DATA_HOME" ]; then
```

Comment:
I didn't read the XDG spec, but my understanding from the issue you linked is we should default to `$HOME/.local/share` if `XDG_DATA_HOME` is unset, instead of returning an error. Should we change this behavior?

Same comment for `XDG_CONFIG_HOME` if the spec defines a fallback for that dir

Reply:
Yes, I can make this change.

Reply:
Changes made.  Added a function `make_xdg_path` to follow the spec which states:

> If, when attempting to write a file, the destination directory is non-existent an attempt should be made to create it with permission 0700. If the destination directory exists already the permissions should not be changed.

I did add a dependency of `realpath` to avoid that function from creating a bunch of directories that could be caused with a bad relative path.  If you would like to remove it, I can.